tangelo/toolboxes/ansatz_generator/uccsd.py

```diff
@@ -262,22 +263,33 @@ def _compute_mp2_params(self):
         Returns:
             list of float: The initial variational parameters.
         """
+        # Circular import so importing here
+        from tangelo.algorithms.classical import MP2Solver
```

Comment:
oooh boy I don't like this one.
Where is this cycle coming from ?

Reply:
```
File "<stdin>", line 1, in <module>
  File "/Users/jamesbrown/Documents/Coding/TangeloForks/Tangelo/tangelo/toolboxes/ansatz_generator/__init__.py", line 20, in <module>
    from .uccsd import UCCSD
  File "/Users/jamesbrown/Documents/Coding/TangeloForks/Tangelo/tangelo/toolboxes/ansatz_generator/uccsd.py", line 37, in <module>
    from tangelo.algorithms import MP2Solver
  File "/Users/jamesbrown/Documents/Coding/TangeloForks/Tangelo/tangelo/algorithms/__init__.py", line 15, in <module>
    from .variational import VQESolver, ADAPTSolver, SA_VQESolver, SA_OO_Solver
  File "/Users/jamesbrown/Documents/Coding/TangeloForks/Tangelo/tangelo/algorithms/variational/__init__.py", line 15, in <module>
    from .adapt_vqe_solver import ADAPTSolver
  File "/Users/jamesbrown/Documents/Coding/TangeloForks/Tangelo/tangelo/algorithms/variational/adapt_vqe_solver.py", line 41, in <module>
    from tangelo.algorithms.variational.vqe_solver import VQESolver
  File "/Users/jamesbrown/Documents/Coding/TangeloForks/Tangelo/tangelo/algorithms/variational/vqe_solver.py", line 37, in <module>
    class BuiltInAnsatze(Enum):
  File "/Users/jamesbrown/Documents/Coding/TangeloForks/Tangelo/tangelo/algorithms/variational/vqe_solver.py", line 39, in BuiltInAnsatze
    UCCSD = agen.UCCSD
AttributeError: partially initialized module 'tangelo.toolboxes.ansatz_generator' has no attribute 'UCCSD' (most likely due to a circular import)
    ```

Reply:
I think we can fix this one by importing at the top instead
`from tangelo.algorithms.classical.mp2_solver import MP2Solver`

this way python does not try to import everything thats inside `tangelo.algorithms` which leads to the circular import.

Would you please try that ?
This suggests that to avoid circular dependencies we should try to use that more often in the future: directly import from the module itself, to avoid importing other things as a side effect

Reply:
This worked